joplin/js/editor.js

```diff
@@ -262,29 +255,113 @@ $(function() {
     }
   }
 
-  var enButton = $('#en');
-  enButton.click(function() {
-    changeLanguage('en');
+  if (localStorage.selected_lang) {
+    state.currentLang = localStorage.selected_lang;
+    updateSelectedLanguage(state.currentLang);
+    changeLanguage(state.currentLang);
+    localStorage.removeItem('selected_lang');
+  } else {
+    state.currentLang = 'en';
+    updateSelectedLanguage(state.currentLang);
+    changeLanguage(state.currentLang);
+  }
+
+  // watch language select for changes
+  $('#language-select').change(function(currentLang) {
+    let selectedLanguage = document.getElementById('language-select')
+      .selectedOptions[0];
+    changeLanguage(selectedLanguage.id);
+            updateSelectedLanguage(state.currentLang);
+    localStorage.selected_lang = state.currentLang;
   });
-  var esButton = $('#es');
-  esButton.click(function() {
-    changeLanguage('es');
+
+  // case function for setting the selected language on dropdown
+  // maybe there is a less verbose way to do this?
+  function updateSelectedLanguage(currentLang) {
+
+    var contentLink = document.getElementsByClassName("tab-nav merged")[0].firstElementChild.firstElementChild
+
+    switch (currentLang) {
+      case 'en':
+        document.getElementById('language-select').value = 'English';
+        contentLink.innerText = document.getElementById(
+          'language-select',
+        ).value;
+        break;
+      case 'es':
+        document.getElementById('language-select').value = 'Spanish';
+        contentLink.innerText = document.getElementById(
+          'language-select',
+        ).value;
+        break;
+      case 'vi':
+        document.getElementById('language-select').value = 'Vietnamese';
+        contentLink.innerText = document.getElementById(
+          'language-select',
+        ).value;
+        break;
+      case 'ar':
+        document.getElementById('language-select').value = 'Arabic';
+        contentLink.innerText = document.getElementById(
+          'language-select',
+        ).value;
+        break;
+    }
+  }
+
+
+
+  // Persist language for preview even after page refreshes on save
+  var previewButton = $('#page-preview-button');
+  previewButton.click(function() {
+    if (localStorage.selected_lang) {
+      changeLanguage(localStorage.selected_lang);
+      updateSelectedLanguage(localStorage.selected_lang);
+      window.open(state.janisPreviewUrl, '_blank');
+    }
   });
-  var arButton = $('#ar');
-  arButton.click(function() {
-    changeLanguage('ar');
+
+  // Persist language for sharing even after page refreshes on save
+  var shareButton = $('#page-share-preview-button');
+  var urlcopied = $('#page-share-url-copied');
+  shareButton.click(function() {
+    // Not quite sure how the state/localstorage stuff is
+    // working here but hopefully this gets us some links
```

Comment:
copypasta from old PR:
based on this note: https://github.com/cityofaustin/techstack/issues/2827#issuecomment-537265221

"Share button generates clipboard link as expected"

This seems to fix it, I'm not sure the logic is perfect.